### PR TITLE
Address some TAG review feedback.

### PR DESCRIPTION
**Pull request by rmcilroy**
Addresses some of #31

### CODE REVIEW

**Comment by rmcilroy** [comment]
@dbaron, @igrigorik, @toddreifsteck, please take a look and let me know if you have any comments, thanks!

**Comment by dbaron** [comment]
@rmcilroy Curious why you dropped the <code>optional</code> from the <code>timeout</code> in <code>IdleRequestOptions</code>.

Also, the "who's <var>timeout</var> occurred before" seems rather less precise than it ought to be; it should probably say explicitly what <code>timeout</code> (if present) is added to.  (Also, "who's" -> "whose".)

**Comment by rmcilroy** [comment]
> @rmcilroy Curious why you dropped the optional from the timeout in IdleRequestOptions.

Apparently 'optional' is not allowed in a dictionary in IDL (dictionary properties are optional by default), so this was fixing an IDL bug (see https://github.com/w3c/requestidlecallback/pull/30#issuecomment-148805790).

> Also, the "who's timeout occurred before" seems rather less precise than it ought to be; it should probably say explicitly what timeout (if present) is added to. (Also, "who's" -> "whose".)

Done in the latest patch.

**Comment by rmcilroy** [comment]
Is this change good to land?

File: index.html [hunk 1]
```diff
@@ -267,7 +267,7 @@
 <li>If the <var>timeout</var> property is present in <var>options</var> and has a positive value:
 <ol type='a'>
   <li>Wait for <var>timeout</var> milliseconds.</li>
```
Comment (igrigorik) [inline index.html:269]: Hmm, should there be a hook in here somewhere to cancel the timer steps if and when the callback is executed before these timers fire?

**Comment by rmcilroy** [comment]
Good point! Added a line in the latest patch to ensure we only call the timeout callback algorithm if the handle is still in the list of callbacks (it would get removed if it was run during an idle period, or explicitly cancelled)

**Comment by rmcilroy** [comment]
Actually, I just realized, the timeout callback algorithm already does this check, so the additional check at that line isn't really necessary. We could either keep both checks (for clarity) or remove on or the other, whichever you think makes things clearer.

**Comment by igrigorik** [comment]
Ahh, I completely missed the point of the first step there.. I guess there is no reason to keep it in both places. Is there any merit in doing that check in the new location, instead of after the task is queued? I'll defer to your judgement here.

**Comment by rmcilroy** [comment]
I removed the change for the timeout check in the requestIdleCallback algorithm - I think it makes more sense to do it where it originally was in the timeout algorithm.

**Comment by igrigorik** [comment]
sg. With that resolved, LGTM. @dbaron @toddreifsteck any other feedback on this?

**Comment by igrigorik** [comment]
@dbaron @toddreifsteck can you guys please review this?

**Comment by toddreifsteck** [comment]
LGTM

**Pull request merged**